askbot/models/__init__.py

```diff
@@ -4109,6 +4113,11 @@ def record_spam_rejection(
     post_anonymous_askbot_content,
     dispatch_uid='post_anon_content_on_login'
 )
+signals.post_save.connect(
```

Comment:
the idea of using a signal is to decouple the logic. So we don't call clear_cached_data or reset_cached_data everywhere, instead we abstract it's calling to a signal that's known to be fired when a thread is saved.

If we can then make sure this signal only is triggered once in a request/response cycle, we'll be on a good path.